pageserver/src/pgdatadir_mapping.rs

```diff
@@ -675,15 +675,31 @@ impl Timeline {
         lsn: Lsn,
         ctx: &RequestContext,
     ) -> Result<HashMap<String, Bytes>, PageReconstructError> {
-        match self.get(AUX_FILES_KEY, lsn, ctx).await {
-            Ok(buf) => match AuxFilesDirectory::des(&buf).context("deserialization failure") {
-                Ok(dir) => Ok(dir.files),
-                Err(e) => Err(PageReconstructError::from(e)),
-            },
-            Err(e) => {
-                // This is expected: historical databases do not have the key.
-                debug!("Failed to get info about AUX files: {}", e);
-                Ok(HashMap::new())
+        if self.get_switch_to_aux_file_v2() {
+            let kv = self
+                .scan(KeySpace::single(Key::metadata_aux_key_range()), lsn, ctx)
+                .await
+                .context("scan")?;
+            let mut result = HashMap::new();
+            for (_, v) in kv {
+                let v = v.context("get value")?;
+                let v = aux_file::decode_file_value(&v).context("value decode")?;
```

Comment:
Do you intend to keep these contexts or are they for debugging only?

Reply:
I want to keep them